src/client.js

```diff
@@ -158,13 +158,31 @@ Client.prototype.connect = function(args) {
 	var irc = slate(stream);
 	identd.hook(stream, username);
 
-	if (args.password) {
-		irc.pass(args.password);
-	}
+
 
 	irc.me = nick;
-	irc.nick(nick);
-	irc.user(username, realname);
+	if (!!args.sasl) {
```

Comment:
With how javascript implicitly typecasts things, it is probably better from a realistic computer science point of view to do this.